backend/src/main/java/capstone/storage/backend/item/ItemController.java

```diff
@@ -0,0 +1,57 @@
+package capstone.storage.backend.item;
+
+import capstone.storage.backend.exceptions.ExceptionMessage;
+import capstone.storage.backend.exceptions.IsNullOrEmptyException;
+import capstone.storage.backend.exceptions.ItemForbiddenRequestException;
+import capstone.storage.backend.exceptions.ItemToDeleteNotFoundException;
+import capstone.storage.backend.item.models.AddItemDto;
+import capstone.storage.backend.item.models.Item;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/items/")
+@RequiredArgsConstructor
+public class ItemController {
+    private final ItemService service;
+
+    @GetMapping
+    public List<Item> getAllItems() {
+        return service.findAll();
+    }
+
+    @PostMapping(value = {"{eanToFind}", ""})
+    @ResponseStatus(HttpStatus.CREATED)
+    public Item saveItem(@PathVariable(required = false) String eanToFind, @RequestBody AddItemDto addItemDto) {
+
+        if (eanToFind == null || service.isNullOrEmpty(addItemDto)) {
+            throw new IsNullOrEmptyException(ExceptionMessage.IS_NULL_OR_EMPTY_EXCEPTION_MESSAGE.toString());
+        }
+        if (!addItemDto.ean().equals(eanToFind)) {
+            throw new ItemForbiddenRequestException(ExceptionMessage.ITEM_FORBIDDEN_REQUEST_EXCEPTION_MESSAGE.toString());
+        }
+        return service.addItem(addItemDto, eanToFind);
+    }
+    @PutMapping("{id}")
+    public ResponseEntity<Item> updateItem(@PathVariable String id, @RequestBody Item itemToUpdate) {
+        if (itemToUpdate.id().equals(id)) {
+            boolean itemExist = service.existById(id);
+            Item updatedItem = service.updateItem(itemToUpdate);
+            return itemExist ? ResponseEntity.status(HttpStatus.OK).body(updatedItem) : new ResponseEntity<>(HttpStatus.CREATED);
```

Comment:
201 liefert üblicherweise auch die Ressource als body zurück.
```suggestion
            return ResponseEntity.status(itemExist ? HttpStatus.OK : HttpStatus.CREATED).body(updatedItem);
```

Siehe RFC 9110:

> The 201 response content typically describes and links to the
   resource(s) created.

https://datatracker.ietf.org/doc/html/rfc9110#section-15.3.2:~:text=The%20201%20response%20content%20typically%20describes%20and%20links%20to%20the%0A%20%20%20resource(s)%20created.

Und Stackoverflow:
https://stackoverflow.com/a/827045/476791